### ISSUE TITLE
Throws an error - $API Request Authorization failed after a Contribution is submitted using a front-end Contribution page.

### ISSUE THREAD

**Issue by agileware-justin** [civicrm/org.civicrm.recentmenu#19]
Throws an error - $API Request Authorization failed after a Contribution is submitted using a front-end Contribution page.

This error occurs intermittently, but often enough to cause donors to complain. Error was particularly prominent when the volume of donations increased.

An error message is shown on the Thank You step of the Contribution workflow. Causing the Donor to then contact the Organisation as it appears that the donation was not processed correctly.

Workaround was to disable the Recent Menu extension.

Back trace below.

```
Oct 28 17:09:31  [debug] $API Request Authorization failed = #0 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/Civi/API/Kernel.php(153): CRM_Core_Error::backtrace("API Request Auth
orization failed", TRUE)
#1 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/Civi/Api4/Generic/AbstractAction.php(249): Civi\API\Kernel->runRequest(Object(Civi\Api4\Generic\BasicGetAction))
#2 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/ext/org.civicrm.recentmenu/recentmenu.php(136): Civi\Api4\Generic\AbstractAction->execute()
#3 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/ext/org.civicrm.recentmenu/recentmenu.php(94): _get_recentmenu_items()
#4 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php(276): recentmenu_civicrm_preProcess("CRM_Financial_Form_Payment", Object(CRM_Financial_Form_Payment))
#5 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook/WordPress.php(136): CRM_Utils_Hook->runHooks((Array:59), "civicrm_preProcess", 2, "CRM_Financial_Form_Payment", Obj
ect(CRM_Financial_Form_Payment), NULL, NULL, NULL, NULL)
#6 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/Civi/Core/CiviEventDispatcher.php(314): CRM_Utils_Hook_WordPress->invokeViaUF(2, "CRM_Financial_Form_Payment", Object(CRM_Financia
l_Form_Payment), NULL, NULL, NULL, NULL, "civicrm_preProcess")
#7 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/vendor/symfony/event-dispatcher/EventDispatcher.php(251): Civi\Core\CiviEventDispatcher::delegateToUF(Object(Civi\Core\Event\Gener
icHookEvent), "hook_civicrm_preProcess", Object(Civi\Core\UnoptimizedEventDispatcher))
#8 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/vendor/symfony/event-dispatcher/EventDispatcher.php(73): Symfony\Component\EventDispatcher\EventDispatcher->callListeners((Array:1
), "hook_civicrm_preProcess", Object(Civi\Core\Event\GenericHookEvent))
#9 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/Civi/Core/CiviEventDispatcher.php(263): Symfony\Component\EventDispatcher\EventDispatcher->dispatch(Object(Civi\Core\Event\Generic
HookEvent), "hook_civicrm_preProcess")
#10 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php(168): Civi\Core\CiviEventDispatcher->dispatch("hook_civicrm_preProcess", Object(Civi\Core\Event\GenericHookEve
nt))
#11 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Utils/Hook.php(455): CRM_Utils_Hook->invoke((Array:2), "CRM_Financial_Form_Payment", Object(CRM_Financial_Form_Payment), NULL
, NULL, NULL, NULL, "civicrm_preProcess")
#12 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/Form.php(714): CRM_Utils_Hook::preProcess("CRM_Financial_Form_Payment", Object(CRM_Financial_Form_Payment))
#13 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/QuickForm/Action/Display.php(76): CRM_Core_Form->buildForm()
#14 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/packages/HTML/QuickForm/Controller.php(203): CRM_Core_QuickForm_Action_Display->perform(Object(CRM_Financial_Form_Payment), "display")
#15 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/packages/HTML/QuickForm/Page.php(103): HTML_QuickForm_Controller->handle(Object(CRM_Financial_Form_Payment), "display")
#16 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/Controller.php(355): HTML_QuickForm_Page->handle("display")
#17 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Utils/Wrapper.php(98): CRM_Core_Controller->run()
#18 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php(292): CRM_Utils_Wrapper->run("CRM_Financial_Form_Payment", NULL, (Array:0))
#19 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php(69): CRM_Core_Invoke::runItem((Array:17))
#20 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm/CRM/Core/Invoke.php(36): CRM_Core_Invoke::_invoke((Array:3))
#21 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-content/plugins/civicrm/civicrm.php(1200): CRM_Core_Invoke::invoke((Array:3))
#22 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-includes/class-wp-hook.php(310): CiviCRM_For_WordPress->invoke(Object(WP))
#23 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-includes/class-wp-hook.php(334): WP_Hook->apply_filters(NULL, (Array:1))
#24 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-includes/plugin.php(565): WP_Hook->do_action((Array:1))
#25 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-includes/class-wp.php(796): do_action_ref_array("wp", (Array:1))
#26 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-includes/functions.php(1335): WP->main("")
#27 /var/www/vhosts/stopburningstuff.org/httpdocs/wp-blog-header.php(16): wp()
#28 /var/www/vhosts/stopburningstuff.org/httpdocs/index.php(17): require("/var/www/vhosts/stopburningstuff.org/httpdocs/wp-blog-header.php")
#29 {main}
```

Agileware Ref: CIVICRM-2185

**Comment by colemanw** [comment]
@agileware-justin can you please review #20

**Comment by agileware-justin** [comment]
Fixed by #20 

Thanks @colemanw
